Sources/Preferences/Localization.swift

```diff
@@ -99,26 +89,42 @@ struct Localization {
 	- Parameter identifier: Identifier of the string to localize.
 	*/
 	static func get(identifier: Identifier) -> String {
-		let locale = Locale.current
-
 		// Force-unwrapped since all of the involved code is under our control.
 		let localizedDict = Localization.localizedStrings[identifier]!
 		let defaultLocalizedString = localizedDict["en"]!
 
-		guard
-			let languageCode = locale.languageCode,
-			let regionCode = locale.regionCode
-		else {
-			return defaultLocalizedString
+		// Iterate through all user-preferred languages until we find one that has a valid language code
+		var preferredLocale: Locale?
+		for localeID in Locale.preferredLanguages {
+			let locale = Locale(identifier: localeID)
+			if locale.languageCode != nil {
+				preferredLocale = locale
+				break
+			}
 		}
 
-		let localeIdentifier = "\(languageCode)-\(regionCode)"
-		if let localizedString = localizedDict[localeIdentifier] {
-			return localizedString
+		// Fall back to the system language if no language code was found (unlikely, but might happen)
+		if preferredLocale == nil {
+			preferredLocale = Locale.current
+		}
+
+		guard let languageCode = preferredLocale!.languageCode else {
+			return defaultLocalizedString
 		}
 
-		if let localizedString = localizedDict[languageCode] {
-			return localizedString
+		// Chinese is the only language where different region codes result in different translations
```

Comment:
Are you sure about that? What are you basing it on? It feels weird hard-coding this. From experience, assumptions about date, time, and localizations are usually wrong.

Reply:
This is based on comparing the dictionary's entries and checking which region code variants are *actually* different.
I admit it does feel weird to special case this, and I'd be happy to change it, but as long as Chinese is the only language where different region codes produce different translations for "Preferences", I'd prefer keeping it this way.

Reply:
There must be some better system APIs we can use that handles the resolution here so we don't have to hard-code it like this. Good enough for now I guess, but I would want to fix this properly in the future.